#Call program with "python -O" O=Optimize directive for non debug mode. (it is not a zero :))
#When executed in Optimized mode, the program will neither output predicted images with bounding box/mask overlays nor it will write a respective json file. Moreover, the debug traces will not print at error event. In this mode json will be output in a variable.

# UZ: After training, the newly trained model should be place in the model folder, it should be renamed to aoc_model.pth 
datasets:
  train_dataset_name: 'aoc_train_dataset'
  test_dataset_name: 'aoc_test_dataset'
  validation_dataset_name: 'aoc_validation_dataset'
  dataset_train_annotation_url: 'https://lncn.ac/aocanntrain'
  dataset_train_images_url: 'https://lncn.ac/aocdatatrain'
  dataset_test_annotation_url: 'https://lncn.ac/aocanntest'
  dataset_test_images_url: 'https://lncn.ac/aocdatatest'
files:
  # pretrained model used as a training base model, if set as empty, the config file will use imagenet trained model as base.
  pretrained_model_file: ''
  # model_file: './model/aoc_tomato_ripeness_151_40k.pth'
  model_file: './model/aoc_strawberry_class_ripeness.pth' #'./model/aoc_model.pth'
  config_file: 'COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml'
  test_metadata_catalog_file: './data/dataset_catalogs/tom_test_metadata_catalog.pkl'
  train_dataset_catalog_file: './data/dataset_catalogs/tom_train_dataset_catalog.pkl'
  train_annotation_file: './data/tomato_dataset/train/annotations/ripeness_class_annotations.json'
  test_annotation_file: './data/tomato_dataset/test/annotations/ripeness_class_annotations.json'
  validation_annotation_file: './data/tomato_dataset/val/annotations/ripeness_class_annotations.json'
  model_url: 'https://lncn.ac/aocmodel'
  meta_catalog_url: 'https://lncn.ac/aocmeta'
  train_catalog_url: 'https://lncn.ac/aoccat'
directories:
  train_image_dir: './data/strawberry_dataset/train/'
  test_image_dir: '/home/ros/fruit_detector_ws/src/aoc_fruit_detector/scripts/data/bag/rgbd' #'./data/strawberry_dataset/test/'
  validation_image_dir: './data/tomato_dataset/val/'
  training_output_dir: './data/training_output/'
  prediction_output_dir: '/home/ros/fruit_detector_ws/src/aoc_fruit_detector/scripts/data'
  prediction_json_dir: '/home/ros/fruit_detector_ws/src/aoc_fruit_detector/scripts/data'
training:
  epochs: 40000
  number_of_classes: 2
  optimizer: 'SGD'
  learning_rate: 0.0025
settings:
  download_assets: false # if assets such as model and datasets should be downloaded
  rename_pred_images: false #rename the predicted images in img_000001.png like format
  segm_masks: true
  bbox: false
  show_orientation: true
  fruit_type: 'strawberry' # currently supported for "strawberry" or "tomato"
  validation_period: 500 # Smaller validation will increase training time. The value is set to have 100 validation during training
  confidence_threshold: 0.5 # removes annotations with confidence score lower than this value
  filename_patterns:
    rgb: "_rgb"
    depth: "_depth" # filename patterns for rgb and depth images

